src/main/scala/path/Path.scala

```diff
@@ -5,33 +5,41 @@ import com.originate.scalypher.action.ReturnAll
 import com.originate.scalypher.action.ReturnDistinct
 import com.originate.scalypher.action.ReturnReference
 import com.originate.scalypher.Query
+import com.originate.scalypher.MatchQuery
+import com.originate.scalypher.MergeQuery
+import com.originate.scalypher.CreateQuery
 import com.originate.scalypher.types.Referenceable
 import com.originate.scalypher.types.ReferenceableMap
-import com.originate.scalypher.where.ReferenceType
+import com.originate.scalypher.where.Reference
 import com.originate.scalypher.where.Where
 
 case class PathWithWhere(path: Path, where: Where) {
```

Comment:
nope, it's actually an intermediate data structure, only there to allow for sugary syntax. nothing depends on it.

things that are referenceable are things that get identifiers in the query and can receive comparisons/assignments/be returned/deleted/created/updated... so it's just path, node, relationship as far as i can tell from the neo4j docs.